clang/lib/AST/TextNodeDumper.cpp

```diff
@@ -1302,6 +1302,8 @@ void TextNodeDumper::dumpBareTemplateName(TemplateName TN) {
     OS << " index " << STS->getIndex();
     if (std::optional<unsigned int> PackIndex = STS->getPackIndex())
       OS << " pack_index " << *PackIndex;
+    if (STS->getFinal())
+      OS << " final";
```

Comment:
as `final` is a keyword in C++ I wonder if we can find a better name for this.  I certainly wouldn't see this in an AST and know what it means.  I wonder if we should find a better name than `final` throughout this patch.  Something like, `FullySugared` or something?  Though open to other/better names.

Reply:
I am open for suggestions, but I'd prefer to change this in a follow up, since the term is existing and this would make the patch much larger.

Reply:
`FullySugared` seems fine to me, unless you have a better idea.  I would be ok with changing the name in a followup, which wouldn't really require review once we agree on a name.